packages/libs/deer/desert/src/assert.rs

```diff
@@ -26,25 +29,33 @@ where
     assert_tokens_with_context(value, tokens, &Context::new());
 }
 
-pub fn assert_tokens_with_context_error<'de, T>(
-    error: &serde_json::Value,
+pub fn assert_tokens_with_context_error<'de, E, T>(
+    error: &E,
     tokens: &'de [Token],
     context: &Context,
 ) where
+    E: Serialize + Debug,
     T: Deserialize<'de> + Debug,
+    Value: PartialEq<E>,
```

Comment:
I haven't seen this sort of syntax before. Is `Value` the `serde_json` `Value` Enum? Why do we need to set a specific bound for it to be `PartialEq` here?

Reply:
Thanks! I need to change this, I initially added this for better diagnostics, but similar fails to display a diff between two different types properly. I need to change this so that the error is "casted" to the error enum instead. The bound is here because

`assert_eq!` requires that the LHS implements PartialEq<Rhs>. Therefore the bound was needed, and not the other way around.

Reply:
reworked in [af34989](https://github.com/hashintel/hash/pull/1516/commits/af34989f34795e2d9d87ab86d5f4c564d49286ab)

![image](https://user-images.githubusercontent.com/7252775/212747263-0c48cf69-20dc-4d1a-9dd7-33854bee2887.png)

Reply:
Looks great!